Installation path as an option

Zettervall opened the pull request:
I added install_path as an option to the install function to let the user be able to specify where njc should download and install the jre. I also had to add the installation path to both executeJar and execute-jar and execute-class-with-cp to be able to use the JRE installed in another location than default. 
Default path is till __dirname.
My use case is that I want to be able to use NJC inside of a electron app. But the user is not allowed to access files inside of the electron package (__dirname).
In my solution installPath is passed around alot and needed for both install and execution. I could not come up with a more tidy solution but please have a look if you can figure out a way to reuse the install_path for both install and execution.

Zettervall committed:
Added an option for the user to provide an installation path to use for the download and installation of the JRE.

NLueg commented:
Hey @Zettervall, thank you very much for your PR!
The change itself looks good to me. But can you have a look into the unit tests?
They're currently failing.

Zettervall committed:
Added test for the default parameters and
and made the existing test assert that
the default params generates a valid url

Zettervall commented:
@NLueg 
Hi,
Thank you for the feedback. I'm kind of new to JS and TS (I'm a java developer) and I didn't realized there were any tests.
I have now fixed the failing test and since I moved were the call to "generateInstallOptions()" is done I also added an test to test that method. The old "default url for no parameters" test is now making sure that the default params can generate a valid URL if "getUrlToCall is passed with the default ones.

NLueg commented:
No problem, thanks for your update! Looks good to me now.
I just published your change with version `1.1.0`!
